test/cloudtest/test_managed_cluster.py

```diff
@@ -70,3 +70,37 @@ def test_managed_cluster_sizing(mz: MaterializeApplication) -> None:
             wait(condition="condition=Ready", resource=compute_pod)
 
     mz.environmentd.sql("DROP CLUSTER sized1 CASCADE")
+
+    mz.testdrive.run(
+        input=dedent(
+            """
+            ! CREATE CLUSTER sizedbad (SIZE="badsize")
+            contains:unknown cluster replica size badsize
+            """
+        ),
+        no_reset=True,
+    )
+
+    mz.environmentd.sql(
+        'ALTER SYSTEM SET ALLOWED_CLUSTER_REPLICA_SIZES="1"',
+        port="internal",
+        user="mz_system",
+    )
+    try:
+        mz.environmentd.sql(
+            'CREATE CLUSTER mzsizetest (SIZE="2")',
+            port="internal",
+            user="mz_system",
+        )
```

Comment:
I'm not very familiar with these tests, but I just want to double check, something is actually asserting that this executes successfully right?

Reply:
It'll raise an internal error, I could catch InternalError and re-raise an assertion error from it, if that's more the style we're shooting for? Either way the test should catch a regression.
```
>>> con.cursor().execute("create cluster bad (size='badsize');")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/justin/work/cloud/venv/lib/python3.12/site-packages/psycopg/cursor.py", line 732, in execute
    raise ex.with_traceback(None)
psycopg.errors.InternalError_: unknown cluster replica size badsize
```

Reply:
Just the default exception is fine, IMO!

Reply:
> Just the default exception is fine, IMO!

Agreed, I just wanted to make sure that some error/exception was raised.